src/script.js

```diff
@@ -0,0 +1,53 @@
+function waitFor(element, eventName) {
+  return new Promise((resolve) => {
+    // eslint-disable-next-line no-shadow
+    const handler = (event) => {
+      if (event.target === element) {
```

Comment:
The logic to check whether the event's target is the same as the element passed to the `waitFor` function is not necessary. The event listener is already attached to the element, so the handler will only be called when the event is triggered on that element. This check can be removed.